File: src/ros2_bridge/mappers/std_msgs/empty.rs
//! Typed mapper for `std_msgs/msg/Empty`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn empty_to_bus(_msg: ros_env::std_msgs::msg::Empty) -> crate::std_msgs::msg::v1::Empty {
    crate::std_msgs::msg::v1::Empty {}
}

pub(crate) fn empty_to_ros(_bus: crate::std_msgs::msg::v1::Empty) -> ros_env::std_msgs::msg::Empty {
    ros_env::std_msgs::msg::Empty {}
}

#[derive(Clone, Copy, Debug, Default)]
pub struct StdMsgsEmptyMapper;

impl TypedTopicMapper for StdMsgsEmptyMapper {
    type Ros = ros_env::std_msgs::msg::Empty;
    type Bus = crate::std_msgs::msg::v1::Empty;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(empty_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(empty_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/std_msgs/empty.py
"""Generated mapper for `std_msgs/msg/Empty`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.std_msgs.msg.v1 import Empty as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def empty_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    pass
    return bus


def empty_to_ros(bus):
    from std_msgs.msg import Empty as RosMsg

    out = RosMsg()
    pass
    return out


class StdMsgsEmptyMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from std_msgs.msg import Empty as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return empty_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return empty_to_ros(bus)
